src/gazebo_ros_link_attacher.cpp

```diff
@@ -1,211 +1,256 @@
+    return false;
+  }
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "Getting BasePtr of " << model2);
+  physics::BasePtr b2 = this->world->ModelByName(model2);
+  if (b2 == NULL) {
+    RCLCPP_ERROR_STREAM(node_->get_logger(), model2 << " model was not found");
     return false;
   }
 
-  bool GazeboRosLinkAttacher::getJoint(std::string model1, std::string link1,
-                                       std::string model2, std::string link2,
-                                       fixedJoint &joint){
-    fixedJoint j;
-    for(std::vector<fixedJoint>::iterator it = this->joints.begin(); it != this->joints.end(); ++it){
-        j = *it;
-        if ((j.model1.compare(model1) == 0) && (j.model2.compare(model2) == 0)
-                && (j.link1.compare(link1) == 0) && (j.link2.compare(link2) == 0)){
-            joint = j;
-            return true;
-        }
-    }
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "Casting into ModelPtr");
+  physics::ModelPtr m1(dynamic_cast<physics::Model *>(b1.get()));
+  fixed_joint.m1 = m1;
+  physics::ModelPtr m2(dynamic_cast<physics::Model *>(b2.get()));
+  fixed_joint.m2 = m2;
+
+  RCLCPP_DEBUG_STREAM(
+    node_->get_logger(), "Getting link: '" << link1 << "' from model: '" << model1 << "'");
+  physics::LinkPtr l1 = m1->GetLink(link1);
+  if (l1 == NULL) {
+    RCLCPP_ERROR_STREAM(node_->get_logger(), link1 << " link was not found");
     return false;
+  }
+  if (l1->GetInertial() == NULL) {
+    RCLCPP_ERROR_STREAM(node_->get_logger(), "link1 inertia is NULL!");
+  } else {
+    RCLCPP_DEBUG_STREAM(
+      node_->get_logger(),
+      "link1 inertia is not NULL, for example, mass is: " << l1->GetInertial()->Mass());
+  }
+  fixed_joint.l1 = l1;
+  RCLCPP_DEBUG_STREAM(
+    node_->get_logger(), "Getting link: '" << link2 << "' from model: '" << model2 << "'");
+  physics::LinkPtr l2 = m2->GetLink(link2);
+  if (l2 == NULL) {
+    RCLCPP_ERROR_STREAM(node_->get_logger(), link2 << " link was not found");
+    return false;
+  }
+  if (l2->GetInertial() == NULL) {
+    RCLCPP_ERROR_STREAM(node_->get_logger(), "link2 inertia is NULL!");
+  } else {
+    RCLCPP_DEBUG_STREAM(
+      node_->get_logger(),
+      "link2 inertia is not NULL, for example, mass is: " << l2->GetInertial()->Mass());
+  }
+  fixed_joint.l2 = l2;
+
+  RCLCPP_DEBUG_STREAM(
+    node_->get_logger(),
+    "Links are: " << l1->GetName() << " and " << l2->GetName());
+
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "Creating revolute joint on model: '" << model1 << "'");
+  fixed_joint.joint = this->physics->CreateJoint("revolute", m1);
+  this->joints_.push_back(fixed_joint);
+
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "Attach");
+  fixed_joint.joint->Attach(l1, l2);
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "Loading links");
+  fixed_joint.joint->Load(l1, l2, ignition::math::Pose3d());
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "SetModel");
+  fixed_joint.joint->SetModel(m2);
+  /*
```

Comment:
I recommend removing this comment and using the readme section _Bugs & Feature Requests_.  Include a subsection there such as 'Konwn Issue" to report about the SetModel method.